plugins/experimental/remap_stats/remap_stats.c

```diff
@@ -92,19 +92,19 @@ stat_add(char *name, TSMgmtInt amount, TSStatPersistence persist_type, TSMutex c
 static char *
 get_effective_host(TSHttpTxn txn)
 {
-  char *effective_url, *tmp;
   const char *host;
   int len;
   TSMBuffer buf;
   TSMLoc url_loc;
 
-  effective_url = TSHttpTxnEffectiveUrlStringGet(txn, &len);
-  buf           = TSMBufferCreate();
+  buf = TSMBufferCreate();
   if (TS_SUCCESS != TSUrlCreate(buf, &url_loc)) {
     TSDebug(DEBUG_TAG, "unable to create url");
+    TSMBufferDestroy(buf);
     return NULL;
   }
-  tmp = effective_url;
+  char *effective_url, *tmp;
```

Comment:
Why are we moving this down here? Since it's stack allocated the compiler won't care that you put it after the `if` above. Really all we want to avoid is making the call to `TSHttpTxnEffectiveUrlStringGet` which will heap alloc the memory pointed to by `effective_url`.